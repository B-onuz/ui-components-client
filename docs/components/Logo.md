# Logo (Component)

Componente utilizado para inserir o logo da aplicação ou da empresa.

## Como utilizar

```js
import Logo from 'ui-components/lib/components/Logo'
```

```jsx
import React from 'react'
import Logo from 'ui-components/lib/components/Logo'
import logoImg from 'ui-components/assets/bonuz_logo.svg'

export const BonuzLogo = () => <Logo width="180" src={logoImg} alt="B-onuz Logo - legal services" />
```

## Styled-system props

- space https://styled-system.com/table#space
- layout https://styled-system.com/table#layout
