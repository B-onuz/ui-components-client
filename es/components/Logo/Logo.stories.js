'use strict'

Object.defineProperty(exports, '__esModule', {
  value: true,
})
exports.LogoExample = exports['default'] = void 0

var _react = _interopRequireDefault(require('react'))

var _Logo = _interopRequireDefault(require('./Logo'))

var _styledComponents = _interopRequireDefault(require('styled-components'))

var _logo_white = _interopRequireDefault(require('../../assets/logo_white.svg'))

var _this = void 0,
  _jsxFileName = '/home/junior/Fontes/ui-components-client/src/components/Logo/Logo.stories.js'

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj }
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(['\n  padding: 18px;\n  width: 180px;\n  background: ', ';\n'])

  _templateObject = function _templateObject() {
    return data
  }

  return data
}

function _taggedTemplateLiteralLoose(strings, raw) {
  if (!raw) {
    raw = strings.slice(0)
  }
  strings.raw = raw
  return strings
}

var _default = {
  title: 'Logo',
  component: _Logo['default'],
}
exports['default'] = _default

var Wrapper = _styledComponents['default'].div(_templateObject(), function (props) {
  return props.theme.colors['primary']
})

var LogoExample = function LogoExample() {
  return /*#__PURE__*/ _react['default'].createElement(
    Wrapper,
    {
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 18,
        columnNumber: 3,
      },
    },
    /*#__PURE__*/ _react['default'].createElement(_Logo['default'], {
      alt: 'B-onuz Logo - legal services',
      width: '180',
      src: _logo_white['default'],
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 19,
        columnNumber: 5,
      },
    })
  )
}

exports.LogoExample = LogoExample
