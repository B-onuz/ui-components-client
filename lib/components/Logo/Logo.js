"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/Logo/Logo.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  display: block;\n  margin: auto;\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var LogoImg = _styledComponents["default"].img(_templateObject(), (0, _styledSystem.compose)(_styledSystem.space, _styledSystem.layout));

var Logo = function Logo(_ref) {
  var children = _ref.children,
      src = _ref.src,
      rest = _objectWithoutPropertiesLoose(_ref, ["children", "src"]);

  return /*#__PURE__*/_react["default"].createElement(LogoImg, _extends({
    src: src
  }, rest, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 12,
      columnNumber: 10
    }
  }));
};

var _default = Logo;
exports["default"] = _default;
module.exports = exports.default;