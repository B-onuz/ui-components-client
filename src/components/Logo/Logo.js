import React from 'react'
import styled from 'styled-components'
import { space, layout, compose } from 'styled-system'

const LogoImg = styled.img`
  display: block;
  margin: auto;
  ${compose(space, layout)}
`

const Logo = ({ children, src, ...rest }) => {
  return <LogoImg src={src} {...rest} />
}

export default Logo
