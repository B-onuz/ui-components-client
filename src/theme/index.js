const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
