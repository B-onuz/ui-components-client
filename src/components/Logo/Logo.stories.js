import React from 'react'
import Logo from './Logo'
import styled from 'styled-components'
import logoImg from '../../assets/logo_white.svg'

export default {
  title: 'Logo',
  component: Logo,
}

const Wrapper = styled.div`
  padding: 18px;
  width: 180px;
  background: ${(props) => props.theme.colors['primary']};
`

export const LogoExample = () => (
  <Wrapper>
    <Logo alt={'B-onuz Logo - legal services'} width="180" src={logoImg} />
  </Wrapper>
)
