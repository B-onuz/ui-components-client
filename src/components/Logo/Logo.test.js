import React from 'react'
import Logo from './Logo.js'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import logoImg from '../../assets/logo_white.svg'
import 'jest-styled-components'

test('Logo changes the class when hovered', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Logo src={logoImg} alt={'B-onuz Logo - legal services'} />
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
