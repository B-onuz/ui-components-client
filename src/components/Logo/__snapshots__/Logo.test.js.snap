// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo changes the class when hovered 1`] = `
.c0 {
  display: block;
  margin: auto;
}

<img
  alt="B-onuz Logo - legal services"
  className="c0"
  src={[Function]}
/>
`;
